### PR TITLE
build(ui): fix report generation & collection for JS tests

### PR DESCRIPTION
**Pull request by danxmoran**
Noticed while working on flaky tests. Will comment explanations inline.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -229,7 +229,6 @@ jobs:
     docker:
       - image: quay.io/influxdb/influxdb-circleci:latest
     working_directory: /go/src/github.com/influxdata/influxdb
-    parallelism: 8
```
Comment (danxmoran) [inline .circleci/config.yml:232]: Our JS tests are so quick, there's not much point to parallelizing across many VMs

File: .circleci/config.yml [hunk 2]
```diff
@@ -245,8 +244,7 @@ jobs:
             # This should usually be a no-op (fully contained in the Yarn cache above), but we
             # include it to be safe since `yarn test` won't auto-install missing modules.
             yarn install --immutable
-            TESTFILES=$(circleci tests glob "src/**/*.test.ts*" | circleci tests split --split-by=timings)
-            yarn test:circleci $TESTFILES
+            yarn test:ci
```
Comment (danxmoran) [inline .circleci/config.yml:247]: `yarn test:circleci` called into `yarn test:ci` with a limit on `maxWorkers`. Now that we have no parallelism, there's no need for the limit.

File: ui/jest.config.js [hunk 3]
```diff
@@ -33,5 +33,5 @@ module.exports = {
   ],
   coverageDirectory: './coverage',
   coverageReporters: ['html', 'cobertura'],
-  reporters: ['default', 'jest-junit'],
+  reporters: ['default', ['jest-junit', { classNameTemplate: '{filepath}' }]],
```
Comment (danxmoran) [inline ui/jest.config.js:36]: Before this change, the JUnit reports generated by `jest` used the test name (free text full of spaces) as the class name. This isn't a perfect replacement, but it's enough to make Circle happy.

Comment (davidby-influx) [reply ui/jest.config.js:36]: We are assuming we're careful not to have spaces in our filepath, which seems reasonable. Only savages do that.

**Review by davidby-influx — APPROVED** [review]
Looks good.

**Review by davidby-influx — APPROVED** [review]

**Pull request merged**